test/runtests.jl

```diff
@@ -106,6 +106,17 @@ T = Float32
         @test isapprox( signalsum, T(2), atol = 5e-3 )
         nt = NamedTuple(sim)
         @test sim == Simulation(nt)
+        # test handling at r = 0 (see PR #322)
+        calculate_weighting_potential!(sim, 4, convergence_limit = 1e-6, device_array_type = device_array_type, refinement_limits = [0.2, 0.1, 0.05], verbose = false)
+        let wp = sim.weighting_potentials[4]
+            φidx1 = SolidStateDetectors.searchsortednearest(wp.grid.φ, T(deg2rad(30)))
+            φidx2 = SolidStateDetectors.searchsortednearest(wp.grid.φ, T(deg2rad(210)))
+            rmin = SolidStateDetectors.searchsortednearest(wp.grid.r, T(0.01))
+            rmax = SolidStateDetectors.searchsortednearest(wp.grid.r, T(0.035))
+            zidx = SolidStateDetectors.searchsortednearest(wp.grid.z, T(0.02))
+            l = vcat(wp.data[rmin:-1:2,φidx2,zidx], wp.data[1:rmax,φidx1,zidx])
+            @test all(diff(l) .< 0) # the weighting potential should monotonously decrease from rmin to rmax
+        end
```

Comment:
I added a very basic test that checks that the weighting potential decreases monotonously along a horizontal line at the middle height of the segmented BEGe test detector:
![image](https://user-images.githubusercontent.com/30291312/193885145-bb35e342-e65f-41f1-98e2-f7b810951b2f.png)
![image](https://user-images.githubusercontent.com/30291312/193885178-b9d8998a-da73-4776-bdc5-e7a63aad346b.png)

Not optimal, but fails with the old version, passes with the new one and should spot obvious bugs in the future.